app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,51 @@
+class FoodsController < ApplicationController
+  def index
+    @foods = Food.perishable
+  end
+
+  def show
+    @food = Food.find(params[:id])
+  end
+
+  def show_by_food_group
+    @food_group = food_group.find(params[:id])
+    @foods = Food.owned_by_food_group(params)
+  end
```

Comment:
For this method, you can simplify line 12 to

`@foods = @food_group.foods`

The reason why we can do this is because of the one to many relationship between food groups and food.